### ISSUE TITLE
Validation Set is not consistent when varying the train ratio.

### ISSUE THREAD

**Issue by DanielChaseButterfield** [lunarlab-gatech/MorphoSymm-Replication#1]
Upon reviewing the code in `umich_contact_dataset.py`, it appears that the validation set changes if the train_ratio changes, even if the val_ratio stays the same.

**Comment by DanielChaseButterfield** [comment]
For now, although this changes the validation set; the validation set is not used for Generating Figures 4-Left or Figures 4-Center, as they use the train and test sets.

Therefore, we can run that part of the experiment without fixing this bug.

Additionally, since the train ratio doesn't vary for Figure 4-right experiment, this also isn't an issue there either, as the validation set will be consistent in that scenario.

**Comment by DanielChaseButterfield** [comment]
Correction, I believe I was wrong in my previous comment. The file `train_supervised.py` uses validation loss (`val_loss`) in order to run the Early Stopping mechanism. This could change the final train losses and test f1-scores of Figures 4-left and center, since the validation set is inconsistent between train ratios.

Therefore, I probably do need to fix this bug in order to replicate that experiment.

However, our trained models of CNN, CNN-aug, and ECNN for 0.85 train ratio should be fine, as they all have the same validation set, and its most likely the last 15% of the dataset.

**Comment by DanielChaseButterfield** [comment]
This was fixed for the Sample Efficiency experiment on the [Sample Efficiency Branch](https://github.com/lunarlab-gatech/MorphoSymm-Replication/tree/rss2023sampleEfficiency).
